commander/new_block.go

```diff
@@ -193,6 +198,25 @@ func (c *Commander) syncRange(startBlock, endBlock uint64) error {
 		return errors.WithStack(err)
 	}
 
+	err = c.syncStakeWithdrawals(startBlock, endBlock)
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func (c *Commander) withdrawRemainingStakes(startBlock, endBlock uint64) error {
+	stakes, err := c.storage.GetPendingStakeWithdrawalsByFinalisationBlock(uint32(startBlock), uint32(endBlock))
+	if err != nil {
+		return err
+	}
+	for _, stake := range stakes {
```

Comment:
Instead of querying for a block range here I would create a function `GetReadyStateWithdrawals(currentBlock)` which would return all `PendingStakeWithdrawals` which have `FinalisationBlock` <= `currentBlock`. Underneath you should forward iterate over the index keys for best performance. You can find an example of such iteration here: https://github.com/Worldcoin/hubble-commander/blob/b802d1d952c3c4cffe476ad8969fea943d291f2c/storage/transfer.go#L64. You can use `db.PrefetchIteratorOpts` to prefetch iterator values.